xdsl/ir/core.py

```diff
@@ -225,10 +226,9 @@ def get_optional_attr(
         # Otherwise, check if the attribute dialect is registered.
         dialect_name, _ = name.split(".", 1)
         if dialect_name in self._registered_dialects:
-            if dialect_name in self._loaded_dialects:
-                return None
-            self.load_registered_dialect(dialect_name)
-            return self.get_optional_attr(name)
+            if dialect_name not in self._loaded_dialects:
+                self.load_registered_dialect(dialect_name)
+                return self.get_optional_attr(name)
```

Comment:
nit: can you add an `and` and unnest like above?

Reply:
Right!